packages/ps/lib/runtime-api.js

```diff
@@ -0,0 +1,277 @@
+  })
+
+  if (statusCode !== 200) {
+    const error = await body.text()
+    throw new Error(`Failed to stop runtime services: ${error}`)
+  }
+}
+
+async function startRuntimeServices (pid) {
+  const socketPath = getSocketPathFromPid(pid)
+  const client = new Client({
+    hostname: 'localhost',
+    protocol: 'http:'
+  }, { socketPath })
+
+  const { statusCode, body } = await client.request({
+    path: '/api/services/start',
+    method: 'POST'
+  })
+
+  if (statusCode !== 200) {
+    const error = await body.text()
+    throw new Error(`Failed to start runtime services: ${error}`)
+  }
+}
+
+async function restartRuntimeServices (pid) {
+  const socketPath = getSocketPathFromPid(pid)
+  const client = new Client({
+    hostname: 'localhost',
+    protocol: 'http:'
+  }, { socketPath })
+
+  const { statusCode, body } = await client.request({
+    path: '/api/services/restart',
+    method: 'POST'
+  })
+
+  if (statusCode !== 200) {
+    const error = await body.text()
+    throw new Error(`Failed to restart runtime services: ${error}`)
+  }
+}
+
+async function closeRuntimeServices (pid) {
+  const socketPath = getSocketPathFromPid(pid)
+  const client = new Client({
+    hostname: 'localhost',
+    protocol: 'http:'
+  }, { socketPath })
+
+  const { statusCode, body } = await client.request({
+    path: '/api/services/close',
+    method: 'POST'
+  })
+
+  if (statusCode !== 200) {
+    const error = await body.text()
+    throw new Error(`Failed to close runtime services: ${error}`)
+  }
+}
+
+function pipeRuntimeLogsStream (pid, options, onMessage) {
+  const socketPath = getSocketPathFromPid(pid)
+  let query = ''
+  if (options.level || options.pretty || options.serviceId) {
+    query = '?' + new URLSearchParams(options).toString()
+  }
+
+  const protocol = platform() === 'win32' ? 'ws+unix:' : 'ws+unix://'
+  const webSocket = new WebSocket(protocol + socketPath + ':/api/logs' + query)
+
+  webSocket.on('error', (err) => {
+    throw new Error(`WebSocket error: ${err.message}`)
+  })
+
+  webSocket.on('message', (data) => {
+    onMessage(data.toString())
+  })
+}
+
+async function injectRuntime (pid, serviceId, options) {
+  const socketPath = getSocketPathFromPid(pid)
+  const client = new Client({
+    hostname: 'localhost',
+    protocol: 'http:'
+  }, { socketPath })
+
+  const response = await client.request({
+    path: `/api/services/${serviceId}/proxy` + options.url,
+    method: options.method,
+    headers: options.headers,
+    query: options.query,
+    body: options.body
+  })
+
+  const body = await response.body.text()
+
+  return {
+    statusCode: response.statusCode,
+    headers: response.headers,
+    body
+  }
+}
+
+module.exports = {
+  getRuntimes,
+  getRuntimeMetadata,
+  getRuntimeServices,
+  injectRuntime,
+  getRuntimeEnv,
+  getRuntimeByPID,
+  getRuntimeByPackageName,
+  stopRuntimeServices,
+  startRuntimeServices,
+  restartRuntimeServices,
+  closeRuntimeServices,
+  pipeRuntimeLogsStream
+}
```

Comment:
This would need to be instatiated, instead of opening a client for each API call.